lib/inferno/dsl/fhir_resource_navigation.rb

```diff
@@ -0,0 +1,199 @@
+require_relative 'primitive_type'
+
+module Inferno
+  module DSL
+    module FHIRResourceNavigation
+      DAR_EXTENSION_URL = 'http://hl7.org/fhir/StructureDefinition/data-absent-reason'.freeze
+      PRIMITIVE_DATA_TYPES = FHIR::PRIMITIVES.keys
+
+      def resolve_path(elements, path)
```

Comment:
Capturing a comment made offline -- we need good documentation for all of these new functions and classes